terraform/scripts/ubuntu-builder.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2023 Technology Innovation Institute (TII)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -x # debug
+
+################################################################################
+
+# Assume root if HOME and USER are unset
+[ -z "${HOME}" ] && export HOME="/root"
+[ -z "${USER}" ] && export USER="root"
+
+################################################################################
+
+apt_update () {
+    sudo apt-get update -y
+    sudo apt-get upgrade -y
+    sudo apt-get install -y ca-certificates curl xz-utils 
+}
+
+install_nix () {
+    type="$1"
+    if [ "$type" = "single" ]; then
+        # Single-user
+        sh <(curl -L https://nixos.org/nix/install) --yes --no-daemon
+    elif [ "$type" = "multi" ]; then
+        # Multi-user
+        sh <(curl -L https://nixos.org/nix/install) --yes --daemon
+    else
+        echo "Error: unknown installation type: '$type'"
+        exit 1
+    fi
+    # Fix https://github.com/nix-community/home-manager/issues/3734:
+    sudo mkdir -m 0755 -p /nix/var/nix/{profiles,gcroots}/per-user/"$USER"
+    sudo chown -R "$USER:nixbld" "/nix/var/nix/profiles/per-user/$USER"
+    # Enable flakes
+    extra_nix_conf="experimental-features = nix-command flakes"
+    sudo sh -c "printf '$extra_nix_conf\n'>>/etc/nix/nix.conf"
+    # https://github.com/NixOS/nix/issues/1078#issuecomment-1019327751
+    for f in /nix/var/nix/profiles/default/bin/nix*; do
+        sudo ln -fs "$f" "/usr/bin/$(basename "$f")"
+    done
+}
+
+configure_builder () {
+    # Add user: nix
+    sudo useradd -m -d /home/nix nix
+    # Allow ssh login with buildfarm public key
+    mkdir -p /home/nix/.ssh
+    echo 'ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIIQwSH1R9xShZg1w5dZjRagYLae0QFBPYT3i80iHW1Ej' >> /home/nix/.ssh/authorized_keys
```

Comment:
Maybe sops?

Reply:
I don't see any reason why we could not have the public key in the repo in clear text. Moreover, decrypting the value from sops would add unnecessary complexity, which I would like to avoid.